proposals/NNNN-resource-attributes-in-dxil-and-spirv.md

```diff
@@ -0,0 +1,386 @@
+this writing these docs don't yet cover the texture types, but they'll follow
+from the constraints described here.
+
+[target extension types]:
+    https://llvm.org/docs/LangRef.html#target-extension-type
+[proposal 0006]:
+    https://github.com/llvm/wg-hlsl/blob/main/proposals/0006-resource-representations.md
+[DXILResources docs]:
+    https://llvm.org/docs/DirectX/DXILResources.html
+    
+### SPIR-V
+
+DXC's SPIR-V backend treats "Buffer" as a texture and maps [DXC textures]
+directly to [OpTypeImage]. In LLVM IR there is a 1-1 correspondence between the
+set of SPIR-V target extension types and the resulting OpTypeImage for the
+simple cases:
+
+[DXC textures]:
+    https://github.com/microsoft/DirectXShaderCompiler/blob/main/docs/SPIR-V.rst#textures
+[OpTypeImage]:
+    https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#OpTypeImage
+
+
+|                                 | Dim    | Depth       | Arrayed | MS    | Sampled | Access |
+|---------------------------------|--------|-------------|---------|-------|---------|--------|
+| Texture1D                       | 1D     | Unknown (2) | false   | false | r/o (1) | r/o    |
+| Texture1DArray                  | 1D     | Unknown (2) | true    | false | r/o (1) | r/o    |
+| Texture2D                       | 2D     | Unknown (2) | false   | false | r/o (1) | r/o    |
+| Texture2DArray                  | 2D     | Unknown (2) | true    | false | r/o (1) | r/o    |
+| Texture2DMS                     | 2D     | Unknown (2) | false   | true  | r/o (1) | r/o    |
+| Texture2DMSArray                | 2D     | Unknown (2) | true    | true  | r/o (1) | r/o    |
+| Texture3D                       | 3D     | Unknown (2) | false   | false | r/o (1) | r/o    |
+| TextureCUBE                     | Cube   | Unknown (2) | false   | false | r/o (1) | r/o    |
+| TextureCUBEArray                | Cube   | Unknown (2) | true    | false | r/o (1) | r/o    |
+| RWTexture1D                     | 1D     | Unknown (2) | false   | false | r/w (2) | r/w    |
+| RWTexture1DArray                | 1D     | Unknown (2) | true    | false | r/w (2) | r/w    |
+| RWTexture2D                     | 2D     | Unknown (2) | false   | false | r/w (2) | r/w    |
+| RWTexture2DArray                | 2D     | Unknown (2) | true    | false | r/w (2) | r/w    |
+| RWTexture2DMS                   | 2D     | Unknown (2) | false   | true  | r/w (2) | r/w    |
+| RWTexture2DMSArray              | 2D     | Unknown (2) | true    | true  | r/w (2) | r/w    |
+| RWTexture3D                     | 3D     | Unknown (2) | false   | false | r/w (2) | r/w    |
+| Buffer                          | Buffer | Unknown (2) | false   | false | r/o (1) | r/o    |
+| RWBuffer                        | Buffer | Unknown (2) | false   | false | r/w (2) | r/w    |
+
+DXC lowers [Constant/Texture/Structured/Byte Buffers] to [OpTypeStruct] in
+SPIR-V with a uniform storage class and various layout decorations. DXC tracks
```

Comment:
The storage class should depend on the version of Vulkan that is targeted.  Starting with Vulkan 1.2 the storage class is StorageBuffer.